awx/main/tests/functional/models/test_notifications.py

```diff
@@ -121,8 +125,26 @@ def check_structure(expected_structure, obj):
         job_serialization = UnifiedJobSerializer(job).to_representation(job)
 
         context = job.context(job_serialization)
-        check_structure(TestJobNotificationMixin.CONTEXT_STRUCTURE, context)
+        self.check_structure(TestJobNotificationMixin.CONTEXT_STRUCTURE, context)
+
+    @pytest.mark.django_db
+    def test_schedule_context(self, job_template, admin_user):
+        schedule = Schedule.objects.create(
+            name='job-schedule',
+            rrule='DTSTART:20171129T155939z\nFREQ=MONTHLY',
+            unified_job_template=job_template
+        )
+        job = Job.objects.create(
+            name='fake-job',
+            launch_type='workflow',
+            schedule=schedule,
+            job_template=job_template
+        )
 
+        job_serialization = UnifiedJobSerializer(job).to_representation(job)
```

Comment:
Stepped through this and verified that a schedule does in fact get rendered in the `job_serialization` and `context`. The `check_structure` helper is what actually caught the type error, so that's working, too : )